x-pack/plugins/painless_lab/public/application/hooks/use_submit_code.ts

```diff
@@ -20,18 +21,19 @@ export const useSubmitCode = (http: HttpSetup) => {
 
   const submit = useCallback(
     debounce(
-      async (code: string, context: string, contextSetup: Record<string, any>) => {
+      async (config: RequestPayloadConfig) => {
         setInProgress(true);
 
         // Prevent an older request that resolves after a more recent request from clobbering it.
         // We store the resulting ID in this closure for comparison when the request resolves.
         const requestId = ++currentRequestIdRef.current;
 
         try {
-          localStorage.setItem('painlessLabCode', code);
-          localStorage.setItem('painlessLabContext', context);
-          localStorage.setItem('painlessLabContextSetup', JSON.stringify(contextSetup));
-          const result = await executeCode(http, buildRequestPayload(code, context, contextSetup));
+          const result = await http.post(`${API_BASE_PATH}/execute`, {
+            // Stringify the string, because http runs it through JSON.parse, and we want to actually
+            // send a JSON string.
+            body: JSON.stringify(buildRequestPayload(config)),
```

Comment:
I think it could be even more clear if we do the double stringification here with the comment. `JSON.stringify(JSON.stringify(...)`

Reply:
Ah, nvm I see we are building this up manually in `buildRequestPayload`.